### PR TITLE
FAI-12397 | Added to GitHub source page_size, timeout and params cleanup

### PR DESCRIPTION
**Pull request by dbruno21**
## Description

Added to GitHub source page_size, timeout and params cleanup

## Type of change
- [ ] Bug fix
- [X] New feature
- [ ] Breaking change

### CODE REVIEW

File: sources/github-source/test/utils.ts [hunk 1]
```diff
@@ -1,16 +1,25 @@
 import {AirbyteLogger, readTestResourceAsJSON} from 'faros-airbyte-cdk';
 
-import {GitHub, GitHubToken} from '../src/github';
+import {
+  DEFAULT_BUCKET_ID,
```
Comment (chalenge) [inline sources/github-source/test/utils.ts:4]: feels like an overkill to me, if we only doing this for tests.

File: sources/github-source/test/utils.ts [hunk 2]
```diff
@@ -34,9 +43,11 @@ export function setupGitHubInstance(
           octokitMock.auditLogs ??
           new ErrorWithStatus(400, 'API not available'),
       },
-      githubConfig.bucket_id,
-      githubConfig.bucket_total,
-      githubConfig.fetch_pull_request_files,
+      githubConfig.bucket_id ?? DEFAULT_BUCKET_ID,
```
Comment (chalenge) [inline sources/github-source/test/utils.ts:46]: Fine to put random numbers here

File: sources/github-source/src/github.ts [hunk 3]
```diff
@@ -174,19 +178,21 @@ export abstract class GitHub {
...
-        page_size: PAGE_SIZE,
-        nested_page_size: PR_NESTED_PAGE_SIZE,
+        page_size: this.pageSize,
+        nested_page_size: this.pageSize,
       }
     );
-    for await (const res of this.wrapIterable(iter, this.timeout)) {
+    for await (const res of this.wrapIterable(iter, this.timeout.bind(this))) {
       this.logger.info(
         `Rate limit info si ${JSON.stringify(res['rateLimit'])}`
```
[the remark this answers is not in the document]
Comment (dbruno21) [reply sources/github-source/src/github.ts]: Already removed by Mati on another PR

File: sources/github-source/src/github.ts [hunk 4]
```diff
@@ -796,11 +802,12 @@ export class GitHubApp extends GitHub {
   ): Promise<GitHub> {
     const baseOctokit = makeOctokitClient(cfg, undefined, logger);
     const github = new GitHubApp(
-      cfg,
       baseOctokit,
       cfg.bucket_id ?? DEFAULT_BUCKET_ID,
       cfg.bucket_total ?? DEFAULT_BUCKET_TOTAL,
-      cfg.fetch_files ?? true,
+      cfg.fetch_pull_request_files ?? DEFAULT_FETCH_PR_FILES,
```
[the remark this answers is not in the document]
Comment (dbruno21) [reply sources/github-source/src/github.ts:885]: Well, we need to pass it in 2 places currently, so better to have defaults centralized.
https://github.com/faros-ai/airbyte-connectors/blob/357ddd430db04a97215174e84a0d1e70f9ecd253/sources/github-source/src/github.ts#L839
https://github.com/faros-ai/airbyte-connectors/blob/357ddd430db04a97215174e84a0d1e70f9ecd253/sources/github-source/src/github.ts#L885

I'll see if I can move this to parent constructor (the abstract class)

File: sources/github-source/test/utils.ts (hunk 1 above)
Comment (dbruno21) [reply sources/github-source/test/utils.ts]: not needed anymore

File: sources/github-source/test/utils.ts (hunk 2 above)
Comment (dbruno21) [reply sources/github-source/test/utils.ts]: not needed anymore

File: sources/github-source/src/github.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (dbruno21) [reply sources/github-source/src/github.ts]: I moved initialization to be centralized in one place here https://github.com/faros-ai/airbyte-connectors/blob/5cb2f7e78b4909a138ff243313d020d9c0a68c5b/sources/github-source/src/github.ts#L87-L94

but still I find it cleaner to have all defaults declared together for easier comparison against spec
https://github.com/faros-ai/airbyte-connectors/blob/5cb2f7e78b4909a138ff243313d020d9c0a68c5b/sources/github-source/src/github.ts#L50-L60

**Comment by dbruno21** [comment]
@chalenge PTAL

**Review by chalenge — APPROVED** [review]

**Pull request merged**